### PR TITLE
Using separate database branches for `preview` deployments

### PR DESCRIPTION
**Pull request by adilansari**

### CODE REVIEW

**Comment by adilansari** [comment]
Build is failing because `api.preview.tigrisdata.cloud` does not have DB branching support yet.

File: tsconfig.json [hunk 1]
```diff
@@ -2,20 +2,20 @@
   "compilerOptions": {
     "experimentalDecorators": true,
     "emitDecoratorMetadata": true,
-    "target": "es2017",
+    "target": "esnext",
```
Comment (adilansari) [inline tsconfig.json:5]: [ES modules](https://www.typescriptlang.org/docs/handbook/esm-node.html#type-in-packagejson-and-new-extensions) for top level await

File: tsconfig.json [hunk 2]
```diff
@@ -2,20 +2,20 @@
...
     "lib": ["dom", "dom.iterable", "esnext"],
     "allowJs": true,
     "skipLibCheck": true,
     "strict": true,
     "forceConsistentCasingInFileNames": true,
     "noEmit": true,
     "esModuleInterop": true,
-    "module": "commonjs",
+    "module": "esnext",
```
Comment (adilansari) [inline tsconfig.json:13]: [ES modules](https://www.typescriptlang.org/docs/handbook/esm-node.html#type-in-packagejson-and-new-extensions) for top level await

File: next.config.mjs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/**
+ * @type {import("next").NextConfig}
+ */
+const nextConfig = {
+  swcMinify: true,
+  webpack(config) {
+    config.experiments = { ...config.experiments, topLevelAwait: true };
```
Comment (adilansari) [inline next.config.mjs:7]: webpack support for top level await is still experimental

**Review by ovaistariq — APPROVED** [review]

**Pull request merged**